src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1344,6 +1345,63 @@ macro_rules! coverage_test_alias {
     };
 }
 
+#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Ord, PartialOrd)]
+pub struct RunMakeSupport {
+    pub compiler: Compiler,
+    pub target: TargetSelection,
+}
+
+impl Step for RunMakeSupport {
+    type Output = PathBuf;
+    const DEFAULT: bool = true;
+
+    fn should_run(run: ShouldRun<'_>) -> ShouldRun<'_> {
+        run.never()
+    }
+
+    fn make_run(run: RunConfig<'_>) {
+        let compiler = run.builder.compiler(run.builder.top_stage, run.build_triple());
+        run.builder.ensure(RunMakeSupport { compiler, target: run.target });
+    }
+
+    fn run(self, builder: &Builder<'_>) -> PathBuf {
+        builder.ensure(compile::Std::new(self.compiler, self.target));
+
+        let cargo = tool::prepare_tool_cargo(
+            builder,
+            self.compiler,
+            Mode::ToolStd,
+            self.target,
+            "build",
+            "src/tools/run-make-support",
+            SourceType::InTree,
+            &[],
+        );
+
+        let mut cargo = Command::from(cargo);
+
+        let mut stage_std_dir =
+            builder.stage_out(builder.compiler(builder.top_stage, builder.config.build), Mode::Std);
+        stage_std_dir.push(self.target.triple);
+        stage_std_dir.push("release");
+
+        let stage_std = stage_std_dir.to_str().unwrap();
+        let mut stage_std_deps_dir = stage_std_dir.clone();
+        stage_std_deps_dir.push("deps");
+        let stage_std_deps = stage_std_deps_dir.to_str().unwrap();
+        cargo.env("RUSTFLAGS", format!("-C prefer-dynamic -L {stage_std} -L {stage_std_deps}"));
```

Comment:
The `-Cprefer-dynamic` is indeed necessary, but are the `-L` still necessary? I would expect all libraries to be looked up in the sysroot.

Reply:
Testing this now.

Reply:
I think you are right that the `-L`s are unnecessary.